cpp/ql/test/library-tests/controlflow/guards/GuardsControl.expected

```diff
@@ -81,10 +77,12 @@
 | test.c:126:12:126:26 | call to test3_condition | true | 126 | 128 |
 | test.c:131:7:131:7 | b | true | 131 | 132 |
 | test.c:137:7:137:7 | 0 | false | 142 | 136 |
-| test.c:138:9:138:9 | i | true | 138 | 139 |
 | test.c:146:7:146:8 | ! ... | true | 146 | 147 |
 | test.c:146:8:146:8 | x | false | 146 | 147 |
-| test.cpp:18:8:18:10 | call to get | false | 20 | 16 |
+| test.cpp:18:8:18:10 | call to get | true | 19 | 19 |
```

Comment:
So taking this result as an example, it looks like both the old and new results are correct (assuming 20 ... 16 means the return and function exit).  Why do we lose the old result?
```
Y* Y::f()
{
  if ( get() ) // line 18
    get()->set();
  return 0;
}
```

Reply:
... oh, `get()` has an implementation and always returns `0`.  So this is to do with unreachable code?

Reply:
```
test.cpp:18:8:18:10 | call to get | false | 20 | 16 |
```
is slightly odd, because the end line (16) is before the starting line (20), but you're probably right that it's the function return. It's also clearly not controlled by the test in the if, because - assume `set()` returns - you're going to end up there no matter whether `get()` on line 18 is true or false.

Line 19 is clearly controlled by `get()` on line 18, because you'll only end up there when get returns true (which it cannot, but that doesn't really matter).